src/chatd.cpp

```diff
@@ -2081,7 +2083,7 @@ void Connection::execCommand(const StaticBuffer& buf)
                     chat.clearHistory();
                     // we were notifying NEWMSGs in result of JOINRANGEHIST, but after reload we start receiving OLDMSGs
                     chat.mServerOldHistCbEnabled = mChatdClient.mKarereClient->isChatRoomOpened(chatid);
-                    chat.getHistoryFromDbOrServer(chat.initialHistoryFetchCount);
+                    chat.getHistoryFromDbOrServer(chat.initialHistoryFetchCount, true);
```

Comment:
I think it would be better to solve this bug by:
```suggestion
                    chat.getHistoryFromServer(-chat.initialHistoryFetchCount);
```
No more changes would be required, and the state variables would remain unchanged (like `mServerOldHistCbEnabled`, which must be `false` if the chatroom is not opened at that moment.

Reply:
I agree with you